config/manager/manager.yaml

```diff
@@ -15,6 +15,7 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: modelmesh-controller
+  namespace: system
```

Comment:
do we want the `modelmesh-controller` in the `system` namespace?

Reply:
@Jooho @tjohnson31415

Reply:
I think that `namespace: system` is the default for a Kustomize config generated by the operator-sdk. When rendered through a call to `kustomize` the namespace is typically changed. In our install scripts, the value for the namespace in the rendered manifest is set by a call to `kustomize edit set namespace` (which modifies the `kustomize.yaml` with a namespace override). All that said, I'm not sure what the best practice is here. We should probably be consistent and have all resources with `namespace: system` or none of them.

After a bit of searching, I think we should remove the namespace from the base manifests. I found that [part of the release steps](https://github.com/kserve/modelmesh-serving/blob/main/docs/release-process.md#generate-release-artifacts-and-publish-the-release) includes running `kustomize build config/default > modelmesh.yaml`. For the resources to not have the `namespace:` field we'd need to remove it from the bases.

Good catch!

Reply:
Thanks @tjohnson31415 

@Jooho -- I will remove the approval until you got the chance to make your changes

Reply:
There are more manifests containing `namespace: system`. Do you think I should delete them too?
By the way, if I execute the following command after I delete the namespace from deployment, it shows the following:
~~~
kustomize build config/default
# Warning: 'vars' is deprecated. Please use 'replacements' instead. [EXPERIMENTAL] Run 'kustomize edit fix' to update your Kustomization automatically.
Error: no matches for Id Deployment.v1.apps/modelmesh-controller.system; failed to find unique target for patch Deployment.v1.apps/modelmesh-controller.system
~~~

Reply:
I delete `namespace: system` in most manifests but I leave it for 2 files
- config/webhook/service.yaml
- config/certmanager/certificate.yaml

It is because the following need the namespace in the `config/default/kustomization.yaml`
~~~
  - fieldref:
      fieldPath: metadata.namespace
    name: CERTIFICATE_NAMESPACE_PLACEHOLDER
    objref:
      group: cert-manager.io
      kind: Certificate
      name: modelmesh-webhook-server-cert
      version: v1
  - fieldref:
      fieldPath: metadata.namespace
    name: SERVICE_NAMESPACE_PLACEHOLDER
    objref:
      kind: Service
      name: modelmesh-webhook-server-service
      version: v1
~~~
I think it should be ok for the 2 files to have the namespace. What do you think?

By the way, kustomize command works now.
~~~
kustomize build config/default
~~~

Reply:
> I think we should remove the namespace from the base manifests.

@tjohnson31415 by the way, what are the base manifests?

Reply:
By "base manifests" or "bases" I am referring to the Kube manifest YAML files in the `config` directory that are used with `kustomize` ("bases" and "overlays" are [concepts in Kustomize](https://kubernetes.io/docs/tasks/manage-kubernetes-objects/kustomization/#bases-and-overlays)). So exactly the files that you have now changed.

The webhook Certificate does throw a wrench in the idea of having the released `modelmesh.yaml` without any `namespace` defined as I suggested... The namespace needs to be known at install time to set the `dnsNames` on the Certificate. 🤔 

@ckadner Do you think it would be a problem to remove `modelmesh.yaml` from the release artifacts?

Reply:
I don’t see `modelmesh.yaml` being used anywhere in scripts or referred to in docs — in none of the repos under `kserve`, and in none of the previous releases — so I don’t think it would be a problem if we no longer produce it as a release artifact

So we could go with `namespace: system` only for the webhook resources that need it as you suggested @tjohnson31415

Reply:
@Jooho -- if we all agree about keeping `namespace: system` only for these 2 files:
- config/webhook/service.yaml
- config/certmanager/certificate.yaml

and no longer build the `modelmesh.yaml` release artifact, could you please update the release process instructions to remove this line:
https://github.com/kserve/modelmesh-serving/blob/e9d9366/docs/release-process.md?plain=1#L144

Reply:
ok I deleted the line.